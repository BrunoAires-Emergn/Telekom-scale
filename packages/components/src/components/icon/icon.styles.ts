import { JssStyle } from 'jss';

export const styles: JssStyle = {
  ':host': {
    height: '24px',
    width: '24px',
    display: 'inline-flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
};
