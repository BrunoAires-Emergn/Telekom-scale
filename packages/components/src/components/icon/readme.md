# t-icon



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description                    | Type                                     | Default     |
| ------------- | -------------- | ------------------------------ | ---------------------------------------- | ----------- |
| `customClass` | `custom-class` | (optional) Tag class           | `string`                                 | `''`        |
| `name`        | `name`         | (optional) Tag theme           | `string`                                 | `undefined` |
| `path`        | `path`         |                                | `string`                                 | `undefined` |
| `styles`      | --             | (optional) Injected jss styles | `StyleSheet<string \| number \| symbol>` | `undefined` |


## Methods

### `updateStyles(newStyle: StyleSheet<string | number | symbol>) => Promise<void>`

Method: updateStyles()

#### Returns

Type: `Promise<void>`




----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
