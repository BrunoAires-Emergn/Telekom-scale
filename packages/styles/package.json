{
    "name": "@telements/styles",
    "version": "0.0.1",
    "main": "dist/index.js",
    "scripts": {
        "build": "tsc -b",
        "postbuild": "node ./dist/export-css.js",
        "setup": "yarn"
    },
    "author": "@eeegor <egor.kirpichev@telekom.de>",
    "license": "MIT",
    "devDependencies": {
        "typescript": "^3.4.5"
    },
    "dependencies": {

    }
}
