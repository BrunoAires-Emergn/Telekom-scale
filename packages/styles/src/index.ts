import { button } from './button';
import { alert } from './alert';
import { card } from './card';

const theme = {
	colors: {
		white: '#fff',
		black: '#000',
		blue: '#00f',
		red: '#f00',
	}
}

export {
	theme,
	alert,
	button,
	card,
}
