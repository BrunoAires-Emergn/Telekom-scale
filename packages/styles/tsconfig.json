{
	"extends": "../tsconfig.base",
	"compilerOptions": {
		"rootDir": "./src",
		"outDir": "./dist",
		"types": [
			"node",
			"jest"
		]
	}
}
