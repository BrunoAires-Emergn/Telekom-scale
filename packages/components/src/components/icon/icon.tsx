import { Component, Prop, h, Host } from '@stencil/core';
import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';
import { styles } from './icon.styles';
import { CssInJs } from '../../utils/css-in-js';
import { StyleSheet } from 'jss';
import Base from '../../utils/base-interface';

@Component({
  tag: 'scale-icon',
  shadow: true,
})
export class Icon implements Base {
  /** (optional) Tag class */
  @Prop() customClass?: string = '';
  /** (optional) Tag theme */
  @Prop() name: string;
  @Prop() path: string;

  /** (optional) Injected jss styles */
  @Prop() styles?: StyleSheet;
  /** decorator Jss stylesheet */
  @CssInJs('Icon', styles) stylesheet: StyleSheet;

  componentWillLoad() {}
  componentWillUpdate() {}

  render() {
    return (
      <Host>
        <style>{this.stylesheet.toString()}</style>
        <svg
          class={this.getCssClassMap()}
          width="24"
          height="24"
          viewBox="0 0 26 26"
        >
          <path d={this.path} stroke="black" fill="transparent" />
        </svg>
      </Host>
    );
  }

  getCssClassMap(): CssClassMap {
    const { classes } = this.stylesheet;
    return classNames(
      classes.icon,
      this.name && this.name,
      this.customClass && this.customClass
    );
  }
}
