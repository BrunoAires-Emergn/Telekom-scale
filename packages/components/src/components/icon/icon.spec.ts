import { newSpecPage } from '@stencil/core/testing';
import { Icon } from './icon';
import { styles } from './icon.styles';
import jss from 'jss';

describe('Icon', () => {
  let element;
  beforeEach(async () => {
    element = new Icon();
    element.stylesheet = jss.createStyleSheet(styles as any);
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Icon],
      html: `<t-icon path="d 10 10">Label</t-icon>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle css classes', () => {
    element.customClass = 'custom';
    expect(element.getCssClassMap()).toContain('custom');

    element.name = 'arrow-left';
    expect(element.getCssClassMap()).toContain('arrow-left');
  });
});
