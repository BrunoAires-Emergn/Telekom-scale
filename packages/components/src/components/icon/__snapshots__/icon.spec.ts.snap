// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should match snapshot 1`] = `
<scale-icon path="m 10 10">
  <mock:shadow-root>
    <style>
      .\\:host-0-0-2 {
  width: 24px;
  height: 24px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
}
    </style>
    <svg height="24" viewBox="0 0 26 26" width="24">
      <path d="m 10 10" fill="transparent" stroke="black"></path>
    </svg>
  </mock:shadow-root>
  Label
</scale-icon>
`;
